### DISCUSSION TITLE
Welcome

### DISCUSSION THREAD

**Discussion by dmayo3** [dmayo3/mocksafe#22] [Announcements]
This is a place to discuss this library if you have any questions or ideas.

**Comment by dkMikLev** [comment]
This is exactly what I was looking for! 
I startet using it yesterday and it is just so much more ergonomic than the build in Mock, even with autospec. 

I'll see if I can find some time to help out and create some prs

Thanks for creating and sharing this

**Reply by dmayo3** [reply]
@dkMikLev Thanks! 

I'm glad you like it. 

I've not been actively working on this for quite a while now and although it works it's still kind of immature.

I got stuck trying to figure out how to handle mocking of properties, which is essential to be able to effectively mock in common cases such as trying to mock the popular `requests` library.

**Reply by dkMikLev** [reply]
I would really love to help out! Could you perhaps create some failing tests showing the current issues?

**Reply by dmayo3** [reply]
@dkMikLev great, cheers 👍

Sorry I'm just back off vacation and recovering from jetlag, so I'll need some time before I can get around to it.

**Reply by dmayo3** [reply]
@dkMikLev hey

Sorry for the delay!

So I tried to get my head back into the project and figure out what the issue was, and I came up with nothing.

For now I guess we can assume it's fine, and just kick the tires on the library to see if it's fit for purpose or still has any rough edges.

Thanks for engaging with the project 👍

**Reply by dmayo3** [reply]
@dkMikLev I managed to figure out what the problem was now

I reproduced it here and there's slightly more detail on the referenced issue if needed:

https://github.com/dmayo3/mocksafe/pull/44
